spring-cloud-consul-discovery/src/test/java/org/springframework/cloud/consul/serviceregistry/ActuatorHealthApplicationStatusProviderTests.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2013-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.consul.serviceregistry;
+
+import java.util.stream.Stream;
+
+import com.ecwid.consul.v1.health.model.Check;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import org.springframework.boot.actuate.health.HealthComponent;
+import org.springframework.boot.actuate.health.HealthEndpoint;
+import org.springframework.boot.actuate.health.Status;
+import org.springframework.cloud.consul.discovery.HeartbeatProperties;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
+import static org.springframework.boot.actuate.health.Status.DOWN;
+import static org.springframework.boot.actuate.health.Status.OUT_OF_SERVICE;
+import static org.springframework.boot.actuate.health.Status.UNKNOWN;
+import static org.springframework.boot.actuate.health.Status.UP;
+
+/**
+ * Unit tests for {@link ActuatorHealthApplicationStatusProvider}.
+ *
+ * @author Chris Bono
+ */
+@ExtendWith(MockitoExtension.class)
+class ActuatorHealthApplicationStatusProviderTests {
+
+	@Mock
+	private HealthEndpoint healthEndpoint;
+
+	@Mock
+	private HeartbeatProperties heartbeatProperties;
+
+	@Mock
+	private HealthComponent healthComponent;
+
+	@InjectMocks
+	private ActuatorHealthApplicationStatusProvider applicationStatusProvider;
+
+	static Stream<Arguments> currentCheckStatusBasedOnHealthStatusArgs() {
+		return Stream.of(arguments(UP, Check.CheckStatus.PASSING), arguments(DOWN, Check.CheckStatus.CRITICAL),
+				arguments(OUT_OF_SERVICE, Check.CheckStatus.CRITICAL), arguments(UNKNOWN, Check.CheckStatus.UNKNOWN));
+	}
+
+	@DisplayName("currentStatus() tests")
+	@ParameterizedTest(name = "{index} ==> health status ''{0}'' should map to check status ''{1}''")
```

Comment:
I :heart: Jupiter parameterized tests output in cases like this.

<img width="668" alt="parameterized-test" src="https://user-images.githubusercontent.com/28907971/103160245-fe7a6500-4797-11eb-8aa8-b2dd219967a1.png">